documentation/src/test/java/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java

```diff
@@ -0,0 +1,514 @@
+			// end::fvh-schema-alternative[]
+			assertThat( result ).containsExactlyInAnyOrder();
+		} );
+	}
+
+	@Test
+	public void encoder() {
+		with( entityManagerFactory ).runInTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+
+			// tag::basic-encoder[]
+			List<List<String>> result = searchSession.search( Book.class )
+					.select( f -> f.highlight( "title" ) )
+					.where( f -> f.match().fields( "title" ).matching( "scandal" ) )
+					.highlighter( f -> f.defaultType().encoder( HighlighterEncoder.HTML ) ) // <1>
+					.fetchHits( 20 );
+			// end::basic-encoder[]
+			assertThat( result ).containsExactlyInAnyOrder(
+					Collections.singletonList(
+							"A <em>Scandal</em> in Bohemia"
+					)
+			);
+		} );
+	}
+
+	@Test
+	public void noMatchSize() {
+		with( entityManagerFactory ).runInTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+
+			// tag::no-match-size[]
+			List<List<String>> result = searchSession.search( Book.class )
+					.select( f -> f.highlight( "description" ) )
+					.where( f -> f.bool()
+							.must( f.match().fields( "title" ).matching( "scandal" ) ) // <1>
+							.should( f.match().fields( "description" ).matching( "scandal" ) ) // <2>
+					)
+					.highlighter( f -> f.fastVector().noMatchSize( 100 ) ) // <3>
+					.fetchHits( 20 );
+			// end::no-match-size[]
+			assertThat( result ).containsExactlyInAnyOrder(
+					Collections.singletonList(
+							"The King of Bohemia engages Holmes to recover an indiscreet photograph showing him with the renowned"
+					)
+			);
+		} );
+	}
+
+	@Test
+	public void fragmentSize() {
+		with( entityManagerFactory ).runInTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+
+			// tag::fragment-size[]
+			List<List<String>> result = searchSession.search( Book.class )
+					.select( f -> f.highlight( "description" ) )
+					.where( f -> f.match().fields( "description" ).matching( "king" )
+					)
+					.highlighter( f -> f.fastVector()
+							.fragmentSize( 50 ) // <1>
+							.numberOfFragments( 2 ) // <2>
+					)
+					.fetchHits( 20 );
+			// end::fragment-size[]
+			assertThat( result ).containsExactlyInAnyOrder(
+					Arrays.asList(
+							"The <em>King</em> of Bohemia engages Holmes to recover an indiscreet",
+							"marriage to a daughter of the <em>King</em> of Scandinavia. In disguise"
+					)
+			);
+		} );
+	}
+
+	@Test
+	public void order() {
+		with( entityManagerFactory ).runInTransaction( entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+
+			// tag::basic-order[]
+			List<List<String>> result = searchSession.search( Book.class )
+					.select( f -> f.highlight( "description" ) )
+					.where( f -> f.bool() // <1>
+							.should( f.match().fields( "description" ).matching( "king" ) )
+							.should( f.match().fields( "description" ).matching( "souvenir" ) ).boost( 10.0f )
```

Comment:
I think there's a mistake here, the boost applies to the top-level predicate and that doesn't seem useful.

Also, let's avoid overly long lines as they don't render nicely in the documentation.

```suggestion
							.should( f.match().fields( "description" ).matching( "king" ) )
							.should( f.match().fields( "description" ).matching( "souvenir" )
									.boost( 10.0f ) )
```